### PR TITLE
Add abstract player and concrete localPlayer

### PR DESCRIPTION
**Pull request by julianweise**
This PR adds a abstract Player class and a inherited concrete LocalPlayer class to the game in order to enable game interaction.

### CODE REVIEW

File: src/tetris/brick/brick.ts [hunk 1]
```diff
@@ -19,6 +19,26 @@ export default class Brick {
...
+
+	public moveRight(): void {
+		// TODO: Implement
+	}
+
+	public drop(): void {
+		// TODO: Implement
+	}
+
+	public rotate(): void{
```
Comment (Johannes-Schneider) [inline src/tetris/brick/brick.ts]: In which direction should the brick be rotated? Clockwise vs Counter-Clockwise

File: src/tetris/brick/brick.ts [hunk 2]
```diff
@@ -19,6 +19,26 @@ export default class Brick {
...
+
+	public moveLeft(): void {
+		// TODO: Implement
+	}
+
+	public moveRight(): void {
+		// TODO: Implement
+	}
+
+	public drop(): void {
```
Comment (Johannes-Schneider) [inline src/tetris/brick/brick.ts]: What exactly should `drop` do?

File: src/tetris/player/localPlayer.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		} else if (this._cursor.space.isDown) {
+			moveOperation = this.drop;
+		} else if (this._cursor.up.isDown) {
+			moveOperation = this.rotate;
+		}
+
+		moveOperation();
+
+		// check whether to apply operation a second time (as bias)
+		if (this._biasEventReceiver.has(BiasEvenType.DuplicateInput)) {
```
Comment (Johannes-Schneider) [inline src/tetris/player/localPlayer.ts]: Please move to an extra method `_applyDuplicateInputBias(moveOperation)` for better readability.

File: src/tetris/player/localPlayer.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			const biasEvent = this._biasEventReceiver.get(BiasEvenType.DuplicateInput) as BiasEventDuplicateInput;
+			if(biasEvent.chance <= Math.random()) {
```
Comment (Johannes-Schneider) [inline src/tetris/player/localPlayer.ts]: Add space after brackets please.

File: src/tetris/player/player.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	//endregion
+
+	//region constructor
+	protected constructor(field: Field) {
+		this._field = field;
+	}
+	//endregion
+
+	//region private members
+	private _field: Field;
```
Comment (Johannes-Schneider) [inline src/tetris/player/player.ts]: This should be either `protected` or there should be a protected getter.

File: src/tetris/scene/mainScene.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ import Field from "tetris/field/field";
 import Player from "tetris/player/player";
 import BrickFactory from "tetris/brick/brickFactory";
 import BiasEngine from "tetris/biasEngine/biasEngine";
+import LocalPlayer from "../player/localPlayer";
```
Comment (Johannes-Schneider) [inline src/tetris/scene/mainScene.ts]: Please use absolute imports.

File: src/tetris/brick/brick.ts (hunk 1 above)
Comment (julianweise) [reply src/tetris/brick/brick.ts]: From my point of view common tetris game only allow one direction rotation (clockwise).

File: src/tetris/brick/brick.ts (hunk 2 above)
Comment (julianweise) [reply src/tetris/brick/brick.ts]: added explanation as comments

File: src/tetris/player/player.ts (hunk 5 above)
Comment (julianweise) [reply src/tetris/player/player.ts]: Discussed it offline. Seems to be fine so far.

File: src/tetris/player/localPlayer.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+			moveOperation = this.moveRight;
+		} else if (this._cursor.down.isDown) {
+			moveOperation = this.moveDown;
+		} else if (this._cursor.space.isDown) {
+			moveOperation = this.drop;
+		} else if (this._cursor.up.isDown) {
+			moveOperation = this.rotate;
+		}
+
+		moveOperation();
```
Comment (Johannes-Schneider) [inline src/tetris/player/localPlayer.ts]: Call the `_applyDuplicateInputBias` method here.

**Pull request merged**